core/src/main/c/share/net.c

```diff
@@ -40,6 +40,27 @@ int set_int_sockopt(int fd, int level, int opt, int value) {
     return setsockopt(fd, level, opt, &value, sizeof(value));
 }
 
+JNIEXPORT jint JNICALL Java_io_questdb_network_Net_setKeepAlive0
+        (JNIEnv *e, jclass cl, jint fd, jint idle_sec) {
+    if (set_int_sockopt(fd, SOL_SOCKET, SO_KEEPALIVE, 1) < 0) {
+        return -1;
+    }
+    #if defined(__linux__) || defined(__FreeBSD__)
```

Comment:
nit: could be `#if defined(__unix__)`

Reply:
but is it really better? is MacOS a Unix? Probably more than Linux.

Reply:
> is MacOS a Unix?

Not from the macro perspective. :)

Anyway, let's keep it as is.